### PR TITLE
Update cmake 3.10.2

### PR DESCRIPTION
**Pull request by hjmjohnson**

### CODE REVIEW

**Comment by leej3** [comment]
Thanks, I'll start looking through this now

**Comment by hjmjohnson** [comment]
FYI:  I'm still wrestling with the continuous integration system for using a newer version of cmake.

**Comment by hjmjohnson** [comment]
@leej3 A series of small single topic commits are much easier to review and fix.

If you have not used "git add -p", it is a great way to create small topic related commits quickly.

I often do "git reset origin/master; git add -p; <.. select topic 1 changes ..>; git commit ; git add -p; < .. select topic 2 changes ..>; git commit ; until all changes are in meaningful patch sets.

Of course good commit messages are essential to facilitate review by external folks. :)

**Comment by hjmjohnson** [comment]
@leej3 NOTE: CI environment fixed for newer version of cmake!!!  If I get your approval, I will rebase and merge today.

**Comment by leej3** [comment]
> @leej3 A series of small single topic commits are much easier to review and fix.
> 
> If you have not used "git add -p", it is a great way to create small topic related commits quickly.
> 
> I often do "git reset origin/master; git add -p; <.. select topic 1 changes ..>; git commit ; git add -p; < .. select topic 2 changes ..>; git commit ; until all changes are in meaningful patch sets.
> 
> Of course good commit messages are essential to facilitate review by external folks. :)

Duly noted. I had seen the -p flag a while back but didn't think to use it in this case. Thanks for the suggestion. I shall make a habit of it.

File: utils/CMakeLists.txt [hunk 1]
```diff
@@ -6,9 +6,9 @@ include_directories(${NIFTI_SOURCE_DIR}/niftilib ${NIFTI_SOURCE_DIR}/znzlib ${NI
 add_executable(${PACKAGE_PREFIX}nifti_stats nifti_stats.c)
 add_executable(${PACKAGE_PREFIX}nifti_tool nifti_tool.c)
 add_executable(${PACKAGE_PREFIX}nifti1_test nifti1_test.c)
-target_link_libraries( nifti_stats ${PACKAGE_PREFIX}niftiio ${PACKAGE_PREFIX}nifticdf)
-target_link_libraries( nifti_tool  ${PACKAGE_PREFIX}niftiio)
-target_link_libraries( nifti1_test ${PACKAGE_PREFIX}niftiio)
+target_link_libraries( nifti_stats PUBLIC ${PACKAGE_PREFIX}niftiio ${PACKAGE_PREFIX}nifticdf)
+target_link_libraries( nifti_tool  PUBLIC ${PACKAGE_PREFIX}niftiio)
```
Comment (leej3) [inline utils/CMakeLists.txt]: nifti_stats is not worth building. nifti1_test should be moved to tests and  the nifti_tool should utility should be built from the src in nifti2 (which is compatible with both all nifti files). Can you confirm this @afni-rickr ?

**Comment by leej3** [comment]
Can we keep track of build requirements  in this doc just to keep everything together as we hash it out:

https://docs.google.com/document/d/1tZLNs_HEFFRxZ8Cj1_f3ddMV2tc6OmxWKMTCZK8WV5s/edit?usp=sharing

**Pull request merged**